pkg/controller/util/clustercidr/controller_utils.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clustercidr
+
+import (
+	"fmt"
+
+	networkingv1alpha1 "k8s.io/api/networking/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/kubernetes/pkg/util/slice"
+)
+
+// CreateClusterCIDRHandler creates clusterCIDR handler.
+func CreateClusterCIDRHandler(f func(ccc *networkingv1alpha1.ClusterCIDR) error) func(obj interface{}) {
+	return func(originalObj interface{}) {
+		ccc := originalObj.(*networkingv1alpha1.ClusterCIDR)
+		if err := f(ccc); err != nil {
+			utilruntime.HandleError(fmt.Errorf("error while processing ClusterCIDR Add/Delete: %w", err))
+		}
+	}
+}
+
+// NeedToAddFinalizer checks if a finalizer should be added to the object.
+func NeedToAddFinalizer(obj metav1.Object, finalizer string) bool {
+	return obj.GetDeletionTimestamp() == nil && !slice.ContainsString(obj.GetFinalizers(),
+		finalizer, nil)
+}
```

Comment:
I'd rather try to have these helpers inside of the controller folder `pkg/controller/nodeipam/ipam/multi_cidr_range_allocator.go` or in the same file that uses it without exporting them, we avoid imports and is not much code

Reply:
Done